File: python/scrabble-score/scrabble_score.py
def score(word: str) -> int:
  word_map = {
    'AEIOULNRST': 1,
    'DG': 2,
    'BCMP': 3,
    'FHVWY': 4,
    'K': 5,
    'JX': 8,
    'QZ': 10,
  }

  new_map = {}

  for key in word_map:
    for c in key:
      new_map[c] = word_map[key]

  return sum(new_map[c] for c in word.upper())

File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
val scrabbleMap = mapOf(1 to "AEIOULNRST", 2 to "DG", 3 to "BCMP", 4 to "FHVWY", 5 to "K", 8 to "JX", 10 to "QZ")

object ScrabbleScore {
  fun scoreWord(word: String): Int {

    return word.toUpperCase().map {
      when (it) {
        in scrabbleMap[1]!! -> 1
        in scrabbleMap[2]!! -> 2
        in scrabbleMap[3]!! -> 3
        in scrabbleMap[4]!! -> 4
        in scrabbleMap[5]!! -> 5
        in scrabbleMap[8]!! -> 8
        else -> 10
      }
    }.sum()
  }
}